core/sys/types/build.rs

```diff
@@ -2,35 +2,65 @@
 //! Builds the FFI type bindings for the common SGX SDK types
 
 use bindgen::{callbacks::ParseCallbacks, Builder};
+use once_cell::sync::Lazy;
+
+// Types that don't have an SGX qualifier.
+//
+// These types are of the form <name> and later typedefed to an `sgx_<name>`.
+//
+// ```C
+//      typedef struct _foo_name {
+//          int a;
+//          float b;
+//      } sgx_foo_name;
+// ```
+//
+// To keep the noise out of the bindings, we use the underlying type and tell
+// bindgen to map directly to `sgx_<name>` version.
+static ALLOWED_UNDERLYING_TYPES: Lazy<Vec<&str>> = Lazy::new(|| {
+    vec![
+        "_status_t",
+        "_target_info_t",
+        "_attributes_t",
+        "_report_t",
+        "_key_request_t",
+    ]
+});
 
 #[derive(Debug)]
 struct Callbacks;
 
 impl ParseCallbacks for Callbacks {
     fn item_name(&self, name: &str) -> Option<String> {
-        match name {
-            "_status_t" => Some("sgx_status_t".to_owned()),
-            _ => None,
+        if ALLOWED_UNDERLYING_TYPES.contains(&name) {
+            Some(format!("sgx{}", name))
+        } else if name.starts_with("_sgx") {
```

Comment:
When would this branch trigger? I don't see any `_sgx` in the allowlist calls below

Reply:
Let me know if you think we should try to document in the `build.rs`.  Since this is core it might justify documenting on top of bindgen docs.
https://rust-lang.github.io/rust-bindgen/allowlisting.html
> If we specify allowlisting rules, then bindgen will only generate bindings to types, functions, and global variables that match the allowlisting rules, or are transitively used by a definition that matches them.

Those `_sgx` types are pulled through transitively